nomt/src/bitbox/beatree/leaf/node.rs

```diff
@@ -101,36 +101,37 @@ impl LeafNode {
 pub struct LeafBuilder {
     leaf: LeafNode,
     index: usize,
-    total_value_size: usize,
+    remaining_value_size: usize,
```

Comment:
I'm not entirely sure why this change is made. Is it to verify that what was declared at the beginning of the leaf construction is respected during the insertion of the values? In this case, it makes total sense

Reply:
since the semantic of this changed (we now provide it up-front and reduce it with each value) the rename felt appropriate to me.

but the key change is in L123